learner_functions.py

```diff
@@ -100,3 +120,50 @@ def generate_and_write_results(pro_data, model_gender, model_msi, gender_labels,
             outfile.write('1\n')
 
     outfile.close()
+
+
+def generate_and_write_results_hard_voting(pro_data, model_gender, model_msi, gender_labels, msi_labels, sample_names, strict=True):
+    gender_predictions = hv.hard_vote(model_gender, pro_data, gender_labels, 'gender')
+
+    msi_predictions = hv.hard_vote(model_msi, pro_data, msi_labels, 'msi')
+    outfile = open('subchallenge_1.csv','w')
+    outfile.write('sample,mismatch\n')
+
+    if strict:
+        for i in range(0,len(msi_labels)):
+            outfile.write(sample_names[i] + ',')
+            if gender_labels[i] == gender_predictions[i] and msi_labels[i] == msi_predictions[i]:
+                outfile.write('0\n')
+            else:
+                outfile.write('1\n')
+    else:
+        for i in range(0, len(msi_labels)):
+            outfile.write(sample_names[i] + ',')
+            if gender_labels[i] != gender_predictions[i] and msi_labels[i] != msi_predictions[i]:
+                outfile.write('1\n')
+            else:
+                outfile.write('0\n')
+    outfile.close()
+
+def generate_and_write_probability_voting(pro_data, model_gender, model_msi, gender_labels, msi_labels, sample_names, strict=True):
+    gender_predictions = hv.hard_vote(model_gender, pro_data, gender_labels, 'gender')
```

Comment:
Is this line supposed to use soft voting instead of hard voting? I can't tell a difference between this and the hard voting function, & I'm wondering if that's what was intended.